src/endpoint/s3/s3_rest.js

```diff
@@ -267,6 +270,86 @@ function authenticate_request(req, res) {
     }
 }
 
+async function authorize_request(req) {
+    if (req.op_name === 'put_bucket') return;
+    const policy_info = req.params.bucket && await req.object_sdk.read_bucket_sdk_policy_info(req.params.bucket);
+    const s3_policy = policy_info.s3_policy;
```

Comment:
List_buckets does not apply a bucket to the request.
This will cause this line to fail and throw an exception (access s3_policy of undefined).

Reply:
checked - if no params.bucket will authorise (for sure no bucket_policy)